### DISCUSSION TITLE
Custom cron script

### DISCUSSION THREAD

**Discussion by Marshal4582** [YetiForceCompany/YetiForce#49] [How to]
Gents, how to properly register custom cron script placed /modules/mycustommodule/crons?
Manually setting record in vtiger_cron_task  giving me weird behavior.

**Comment by Marco31Bike** [comment]
Observing what happen here for a while and I believe yetiforce going to ditch community soon. YF7 seems to be evidence (license). Sad.

To Marshal4582:

Depending on your installation and environment you will need to follow PSR4

Make sure your cron script starts with:

namespace Modules\YOURMODULE\Crons; //Watch letter case - important to capitalize it
require_once '/var/www/html/include/main/WebUI.php'; // Ensure Yetiforce environment is loaded
use App\CronHandler;
use Vtiger_Record_Model;
use App\Db;

your class starts here{
}

In this case in 'vtiger_cron_task' table, handler_class column should get namespace path with class: Modules\Content\Crons\YOUR_CLASS_FROM_CRON_SCRIPT

rebuild stuff: composer dump-autoload
restart server (apache,ngix)

Cheers!
